From 17037146bf959f4548f1cf55a55cafe278209a34 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Wed, 5 Apr 2023 14:12:24 +0300
Subject: [PATCH 08/28] [libunwind] Increase the external value of
 _LIBUNWIND_CURSOR_SIZE

For x86_64 Windows targets (that use SEH), _LIBUNWIND_CURSOR_SIZE
is 204.

If the libunwind.h header is included without defining
_LIBUNWIND_IS_NATIVE_ONLY (like in the libunwind test cases), the
sizes are set to accommodate the maximum possible cursors and
contexts.

(Alternatively, __libunwind_config.h should be changed to default
to native unwinding unless cross unwinding has been requested.
Cross unwinding isn't implemented as far as I know anyway.)
---
 libunwind/include/__libunwind_config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libunwind/include/__libunwind_config.h b/libunwind/include/__libunwind_config.h
index a4e1a3613bbf..a317a36677d3 100644
--- a/libunwind/include/__libunwind_config.h
+++ b/libunwind/include/__libunwind_config.h
@@ -198,7 +198,7 @@
 # define _LIBUNWIND_TARGET_S390X 1
  #define _LIBUNWIND_TARGET_LOONGARCH 1
 # define _LIBUNWIND_CONTEXT_SIZE 167
-# define _LIBUNWIND_CURSOR_SIZE 179
+# define _LIBUNWIND_CURSOR_SIZE 204
 # define _LIBUNWIND_HIGHEST_DWARF_REGISTER 287
 #endif // _LIBUNWIND_IS_NATIVE_ONLY
 
-- 
2.34.1

